File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        # Initialize the maximum reachable index
        max_reach = 0

        # Iterate through the array
        for i, jump in enumerate(nums):
            # If the current index cannot be reached, return False
            if i > max_reach:
                return False

            # Update the maximum reachable index
            max_reach = max(max_reach, i + jump)

        # If the last index can be reached, return True
        return max_reach >= len(nums) - 1

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int reachableIndex = 0;

        for (int i = 0; i < nums.length; i++) {
            if (i > reachableIndex) {
                return false;
            }
            reachableIndex = Math.max(reachableIndex, i + nums[i]);
        }

        return true;
    }
}